source/kraken/apply_disruption.cpp

```diff
@@ -579,7 +579,15 @@ struct delete_impacts_visitor : public apply_impacts_visitor {
 
         for(const auto& wptr: modified_by_moved) {
             if (auto share_ptr = wptr.lock()){
-                disruptions_collection.insert(share_ptr);
+                // Do not reapply the same disruption. If we have more than one impact we will
+                // reapply the one we just deleted.
+                // We need to keep the link though, because other impacts from the disruption may
+                // need to be deleted after this one and they need to stay in modified_by.
```

Comment:
Can you add a word about the fact that it is possible because a disruption is always completely deleted (all of its impacts also)?

Reply:
I've added something, I hope it's clear enough. I struggled a bit to explain this in French so in English it's not really easier :D.